File: C++/Day11/eAttendance/src/employee.cpp
#include "../include/employee.h"
using namespace NPerson::NEmployee;

Employee::Employee()
{
}
Employee::Employee(int empid, double salary, string name) : Person(name)
{
    this->empid = empid;
    this->salary = salary;
}
void Employee::acceptData()
{
    cout << "Enter empid = ";
    cin >> this->empid;

    Person::acceptData();

    cout << "Enter salary = ";
    cin >> this->salary;
}
void Employee::displayData()
{
    cout << "Empid = " << this->empid << endl;

    Person::displayData();

    cout << "Salary = " << this->salary << endl;
    cout << "--------------------------" << endl;
}

int Employee::getEmpid()
{
    return this->empid;
}
double Employee::getSalary()
{
    return this->salary;
}

void Employee::editSalary()
{
    cout << "Enter the new Salary = ";
    cin >> this->salary;
}

void NPerson::NEmployee::save(vector<Employee *> &employeeList)
{
    ofstream fout("../data/Employee.txt");

    for (int i = 0; i < employeeList.size(); i++)
    {
        fout << employeeList.at(i)->getEmpid() << ","
             << employeeList.at(i)->getName() << ","
             << employeeList.at(i)->getSalary() << endl;
    }

    fout.close();
}

void NPerson::NEmployee::load(vector<Employee *> &employeeList)
{
    ifstream fin("../data/Employee.txt");
    string data;
    while (getline(fin, data))
    {
        stringstream line(data);
        string token[3];
        for (int i = 0; i < 3; i++)
            getline(line, token[i], ',');
        employeeList.push_back(new Employee(stoi(token[0]), stod(token[2]), token[1]));
    }

    fin.close();
}
Employee *NPerson::NEmployee::find(vector<Employee *> &employeeList)
{
    int id;
    cout << "Enter empid to search = ";
    cin >> id;
    for (int i = 0; i < employeeList.size(); i++)
        if (employeeList.at(i)->getEmpid() == id)
            return employeeList.at(i);
    return NULL;
}

File: CoreJava/Day06/Demo04/src/com/sunbeam/Employee.java
package com.sunbeam;

import java.util.Scanner;

// Employee is-a Person
public class Employee extends Person {
	private int empid;
	private double salary;

	public Employee() {
	}

	public Employee(int empid, String name, double salary) {
		super(name);
		this.empid = empid;
		this.salary = salary;
	}

	@Override
	public void acceptData() // method overriding
	{
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter empid - ");
		this.empid = sc.nextInt();

		// this.acceptData();
		super.acceptData();

		System.out.print("Enter Salary - ");
		this.salary = sc.nextDouble();
	}

	@Override
	public void displayData() // method overriding
	{
		
		System.out.println("Empid - " + this.empid);

		// this.displayData();
		super.displayData();

		System.out.println("Salary - " + this.salary);
	}
	
	public double getSalary() {
		return salary;
	}

}
